build.go

```diff
@@ -108,24 +111,21 @@ func Build(buildManager BuildManager, clock chronos.Clock, logger scribe.Logger)
 	}
 }
 
-func layerFlags(entries []packit.BuildpackPlanEntry) []packit.LayerType {
-	var flags []packit.LayerType
-
+func setLayerFlags(layer packit.Layer, entries []packit.BuildpackPlanEntry) packit.Layer {
 	for _, entry := range entries {
 		launch, ok := entry.Metadata["launch"].(bool)
 		if ok && launch {
-			flags = append(flags, packit.LaunchLayer)
-			flags = append(flags, packit.CacheLayer)
+			layer.Launch = entry.Metadata["launch"] == true
+			layer.Cache = entry.Metadata["launch"] == true
 		}
 	}
 
 	for _, entry := range entries {
 		build, ok := entry.Metadata["build"].(bool)
 		if ok && build {
-			flags = append(flags, packit.BuildLayer)
-			flags = append(flags, packit.CacheLayer)
+			layer.Build = entry.Metadata["build"] == true
+			layer.Cache = entry.Metadata["build"] == true
```

Comment:
Why double checking for the same value?
```suggestion
			layer.Build = true
			layer.Cache = true
```